enhancements/microshift/microshift-default-csi-plugin.md

```diff
@@ -0,0 +1,371 @@
+runtime.  The cluster component versions correlate to the version of OCP that that version of MicroShift is based on. Maps
+of these version correlations are produced as part of the OpenShift CI/CD as container image called the ocp-registry. 
+The ocp-registry is tagged according to the OCP version it represents, and contains the digests of all core OCP components
+associated with that particular version of OCP.
+
+Similarly, ODF maintains a registry image called ocs-registry, which is also tagged according to the OCP version it
+represents, and contains a SQL database of images and their digests to be distributed with that version of OCP.
+
+MicroShift leverages these registry images to ensure parity of its cluster components with the upstream OCP version it is
+derived from.  
+
+Additionally, as MicroShift deploys as a single node, version skew between nodes is not a concern. 
+
+### Operational Aspects of API Extensions
+
+#### Failure Modes
+
+##### Invalid User-Defined LVMD Configuration
+
+The ODF-LVM node pod (`topolvm-node`), validates configuration on start.  Invalid configurations are treated as fatal
+errors and cause the pod to enter a crash loop.  Users can then query the pod's logs to a concise explanation of the 
+configuration error.  See [Variation](#variation) section for steps to correct this state. 
+
+##### Volume Group Does Not Exist
+
+ODF-LVM will verify that LVM Volume Groups specified in the lvmd.yaml file exist at start up.  If any of the volume
+groups cannot be found, the `topolvm-node` pod will enter a crash loop. Users can then query the pod's logs to a concise
+explanation of the configuration error.  See [Variation](#variation) section for steps to correct this state.
+
+##### No Space Left in VolumeGroup
+
+Storage requests will not be fulfilled if the target volume group does not have the capacity to fulfill the requested
+amount.  Remaining capacity is calculated per volume group as the _total unallocated space_ - _spare-gb_ = _remaining
+capacity_.  PVC's with a requests greater than _remaing-capacity_ will not be provisioned. Users may specify the
+_spare-gb_ in the lvmd configuration, per volume group.
+
+>NOTE! ODF-LVM rounds storage request sizes up to the nearest Gib integer value.  This may result in a size greater than
+> the volume groups unallocated space.
+
+This situation presents as PVCs remaining in a "Pending" state indefinitely.  Users should use `oc describe` to examine
+the PVC's events for a message indicating the volume group does not have adequate storage for the request.
+
+#### Support Procedures
+
+## Implementation History
+
+## Alternatives
+
+[KubeVirt CSI Hostpath Provisioner](https://github.com/kubevirt/hostpath-provisioner#kubevirtiohostpath-provisioner) 
+(KV-CSI-HPP) was also considered as a production-grade storage solution.  While it demonstrated a lower resource overhead
+and smaller over-the-wire size, it did not satisfy requirements for workload data isolation.
+
```

Comment:
We also considered exposing only part of the lvmd.conf schema using MicroShift's configuration file.